### ISSUE TITLE
[Bug][Numpy] zeros_like, ones_like does not respect dtype flag

### ISSUE THREAD

**Issue by sxjscience** [apache/incubator-mxnet#16646]
```python
import mxnet as mx
mx.npx.set_np()

a = mx.np.ones((10,10), dtype=mx.np.float32)
b = mx.np.zeros_like(a, dtype=mx.np.int32)
print(b.dtype)
```
Returns
```
float32
```

**Comment by xidulu** [comment]
I take a brief look at the `infertype` function for `zeros_like`:
https://github.com/apache/incubator-mxnet/blob/master/src/operator/numpy/np_init_op.cc#L103
The parameter type seems to be ignored.
The following experiment verified my conjecture:
```
In [1]: from mxnet import np

In [2]: a = np.ones(10)

In [3]: b = np.zeros_like(a, dtype="Life is short, use Pytorch")

In [4]: b
Out[4]: array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0.])

In [5]: b.dtype
Out[5]: dtype('float32')
```

**Comment by Alicia1529** [comment]
This bug has been fixed in pr #16804
